File: solutions/python/grains/1/grains.py
"""Counting grains on a chess board."""

def square(number):
    """Function that counts the number of grains on a square."""
    if number < 1 or number > 64:
        raise ValueError("square must be between 1 and 64")
    return 2**(number-1)

def total():
    """Function that returns total number of grains on a board."""
    counter = 0
    for num in range(1,65):
        counter += square(num)
    return counter
    

File: solutions/c/grains/1/grains.c
#include "grains.h"
#include <math.h>

uint64_t square(uint8_t index){
    if (index > 64 || index < 1){
        return 0;
    }
    return (uint64_t) pow(2,index-1);
}

uint64_t total(void){
    uint64_t sum = 0;
    for(uint64_t i = 1; i <= 64; i++){
        sum += square(i);
    }
    return sum;
}